transformations/transformations/pool_allocator.py

```diff
@@ -530,11 +544,29 @@ def inject_pool_allocator_into_calls(self, routine, targets):
         """
         call_map = {}
         stack_var = self._get_local_stack_var(routine)
+        intfs = FindNodes(Interface).visit(routine.spec)
         for call in FindNodes(CallStatement).visit(routine.body):
             if call.name in targets:
+               # If call is declared via an explicit interface, the ProcedureSymbol corresponding to the call is the
+               # interface block rather than the Subroutine itself. This means we have to update the interface block
+               # accordingly
+                if call.name in [s for i in intfs for s in i.symbols]:
```

Comment:
This list comprehension is potentially performed relatively often. Since `intfs` itself isn't required, this comprehension could be done once, e.g.:
```
intfs_symbols = [s for i in FindNodes(Interface).visit(routine.spec) for s in i.symbols]
```

Reply:
Ah yes, the performance benefits of list comprehensions. Sure, will update, thanks!